### PR TITLE
[IOTDB-2833] Add readme to consensus

### PR DESCRIPTION
**Pull request by SzyWilliam**
Add README.md to consensus package, including the overview and usage.

### CODE REVIEW

File: consensus/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+* `IStateMachine` is the user application that manages a local copy of data.
+* `Peer` is the smallest consensus unit (inside a process), which holds a `IStateMachine` internally.
+* `ConsensusGroup` is a group of `Peer` all managing the same copy of data.
+* `IConsensus` interface defines the basic functionality provided by Consensus Layer. 
+
+User application can create a `ConsensusGroup` with k `Peer` to store data, 
+i.e. that there will be k copies of data.
+
+When write data into a `ConsensusGroup` using `IConsensus::write`, it will be sent to the group leader's `IStateMachine::write` . 
+The leader makes decision about this write operation,
```
Comment (OneSizeFitsQuorum) [inline consensus/README.md:43]: It seems that the leader makes the decision first and then applies write operation to the state machine

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Comment by MrQuansy** [comment]
LGTM

File: consensus/README.md (hunk 1 above)
Comment (SzyWilliam) [reply consensus/README.md:43]: okay, I'll fix the term.

**Pull request merged**